### PR TITLE
82164 - create webjob for bulkprice event

### PR DESCRIPTION
**Pull request by mayur1501364**
1. Created new  UKHO.ERPFacade.PublishPriceChange.WebJob project skeleton
2. Renamed UKHO.ERPFacade.WebJob to UKHO.ERPFacade.Monitoring.WebJob
3. Added required packages and references 
4. Added SetUpFixture for FT
5. Added test filters for UoS and Price Change end point

### CODE REVIEW

**Review by shrey1501620 — APPROVED** [review]

**Review by jrippington — APPROVED** [review]
Looks good

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/WebhookEndpoint.cs [hunk 1]
```diff
@@ -9,20 +9,20 @@ namespace UKHO.ERPFacade.API.FunctionalTests.Helpers
...
+        //public static Config Config;
         private readonly RestClient client;
         private readonly RestClient client2;
         private readonly ADAuthTokenProvider _authToken;
         private SAPXmlHelper SapXmlHelper { get; set; }
 
         public WebhookEndpoint()
         {
-            config = new();
+            //config = new();
```
Comment (RaviK2233) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/WebhookEndpoint.cs:20]: need to remove this comment as we won't use it anymore

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/WebhookEndpoint.cs [hunk 2]
```diff
@@ -9,20 +9,20 @@ namespace UKHO.ERPFacade.API.FunctionalTests.Helpers
 {
     public class WebhookEndpoint
     {
-        public static Config config;
+        //public static Config Config;
```
[the remark this answers is not in the document]
Comment (Sadha1501493) [reply tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/WebhookEndpoint.cs:12]: Done

**Review by RaviK2233 — COMMENTED** [review]
@Sadha1501493  I have few comments, could you please have a look, thanks

File: tests/UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/FTSetupClass.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    
+    [SetUpFixture]
+    internal class FTSetupClass
+    {
+        
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            Config.ConfigSetup();
+            Console.WriteLine("Inside FTSetupClass");
```
[the remark this answers is not in the document]
Comment (Sadha1501493) [reply tests/UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/FTSetupClass.cs]: Removed

**Review by Sadha1501493 — APPROVED** [review]

File: src/UKHO.ERPFacade.PublishPriceChange.WebJob/Program.cs [hunk 4]
```diff
@@ -97,7 +97,7 @@ private static void ConfigureServices(IServiceCollection serviceCollection, ICon
             //create the logger and setup of sinks, filters and properties	
             Log.Logger = new LoggerConfiguration()
                 .MinimumLevel.Information()
-                .WriteTo.File("Logs/UKHO.ERPFacade.WebJob-.txt", rollingInterval: RollingInterval.Day, outputTemplate: "{Timestamp:yyyy-MM-dd HH:mm:ss} [{Level}] [{SourceContext}] {Message}{NewLine}{Exception}")
+                .WriteTo.File("Logs/UKHO.BulkPrice.WebJob-.txt", rollingInterval: RollingInterval.Day, outputTemplate: "{Timestamp:yyyy-MM-dd HH:mm:ss} [{Level}] [{SourceContext}] {Message}{NewLine}{Exception}")
```
Comment (Shirin14926) [inline src/UKHO.ERPFacade.PublishPriceChange.WebJob/Program.cs:100]: Can this be renamed as PublishPriceChange rather than BulkPrice

**Review by Shirin14926 — COMMENTED** [review]
One minor comment please

File: src/UKHO.ERPFacade.PublishPriceChange.WebJob/Program.cs (hunk 4 above)
Comment (mayur1501364) [reply src/UKHO.ERPFacade.PublishPriceChange.WebJob/Program.cs]: Renamed to PublishPriceChange

**Review by Shirin14926 — APPROVED** [review]
LGTM. Thanks @mayur1501364 for changes

**Review by RaviK2233 — APPROVED** [review]
Reviewed FT and looks good.

**Comment by RaviK2233** [comment]
@Sadha1501493 , checks have failed, could you please have a look

**Review by martyn-fewtrell — APPROVED** [review]
Looks good

**Pull request merged**